### PR TITLE
feat: implement path_remove_directory and path_unlink_file system calls

### LINKED ISSUE #49: feature request: Implement both `path_remove_directory` and `path_unlink_file`
# Feature Request

We are implementing 2 of the syscalls that deal with removing files and empty directories.

https://github.com/bjorn3/browser_wasi_shim/blob/7349f7d32a8e6db7ecbd52df8d7373eb0a876130/src/fd.ts#L99-L101
https://github.com/bjorn3/browser_wasi_shim/blob/7349f7d32a8e6db7ecbd52df8d7373eb0a876130/src/fd.ts#L108-L110
https://wasix.org/docs/api-reference/wasi/path_unlink_file
https://wasix.org/docs/api-reference/wasi/path_remove_directory

## Status

We have implemented both and it seems to work for empty file and empty directory.
The code is based on https://github.com/bjorn3/browser_wasi_shim/pull/42/files#diff-177a21b9f40c72ea679cf8658d14ed0d9702afcea694e92e0936f11ab182d2a3R331-R333
with some additional checks.

## Challenges

- Fails on a directory that is not empty. In Golang with https://pkg.go.dev/os#RemoveAll it uses a recursive algorithm with `unlinkat` to unlink the files and remove the empty directories. This goes into an infinite loop.
- The WASI spec requires that we check file and directory permissions/rights before removing/unlinking but the rights base isn't stored anywhere? We couldn't find many examples of checking rights https://github.com/bjorn3/browser_wasi_shim/blob/7349f7d32a8e6db7ecbd52df8d7373eb0a876130/src/wasi_defs.ts#L116
other than https://github.com/bjorn3/browser_wasi_shim/blob/7349f7d32a8e6db7ecbd52df8d7373eb0a876130/src/fs_fd.ts#L290-L291

### PR DESCRIPTION
**Pull request by Prakhar-Agarwal-byte**
Fixes #49 

This adds support for the path_remove_directory and path_unlink_file system calls.

### CODE REVIEW

**Comment by Prakhar-Agarwal-byte** [comment]
We have tested the `path_unlink_file`, and it works as it should but `path_remove_directory `is getting stuck in an infinite loop, will fix that soon.

Would love to get some feedback on this approach.

**Comment by Prakhar-Agarwal-byte** [comment]
@bjorn3 I have fixed the bug, its working fine now

File: src/fs_fd.ts [hunk 1]
```diff
@@ -314,6 +314,47 @@ export class OpenDirectory extends Fd {
...
+  path_unlink_file(path: string): number {
+    const parentDirEntry = this.dir.get_parent_dir_for_path(path);
+    const pathComponents = path.split("/");
+    const fileName = pathComponents[pathComponents.length - 1];
+    const entry = this.dir.get_entry_for_path(path);
+    if (entry === null) {
+      return wasi.ERRNO_NOENT;
+    }
+    if (entry.stat().filetype === wasi.FILETYPE_DIRECTORY) {
+      console.log("file is actually a directory");
```
Comment (bjorn3) [inline src/fs_fd.ts:327]: Please replace this and the other `console.log` below with `debug.log`.

File: src/fs_core.ts [hunk 2]
```diff
@@ -122,6 +122,28 @@ export class Directory {
     return entry;
   }
 
+  get_parent_dir_for_path(
+    path: string,
+  ): File | Directory | SyncOPFSFile | null {
```
Comment (bjorn3) [inline src/fs_core.ts:127]: This function can only return a `Directory` or `null`, right?

**Comment by bjorn3** [comment]
Thanks!

**Pull request merged**